### PR TITLE
Support chunk parallel

### PR DESCRIPTION
**Pull request by Chiiizzzy**
Signed-off-by: Kaiyuan Hu <kaiyuan.hu@zilliz.com>

### CODE REVIEW

File: towhee/functional/mixins/parallel.py [hunk 1]
```diff
@@ -233,11 +235,17 @@ def inner():
...
-                    loop.run_in_executor(executor, _map_task(x, unary_op)))
+                    loop.run_in_executor(executor, self._map_task(x, unary_op)))
+            # for x in self:
+            #     if len(buff) == num_worker:
+            #         queue.put(await buff.pop(0))
+            #     buff.append(
+            #         loop.run_in_executor(executor, self._map_task(x, unary_op)))
             while len(buff) > 0:
                 queue.put(await buff.pop(0))
             queue.put(EOS())
```
Comment (reiase) [inline towhee/functional/mixins/parallel.py:251]: the return value of pmap also needs to be changed.

File: towhee/functional/mixins/parallel.py [hunk 2]
```diff
@@ -233,11 +235,17 @@ def inner():
...
             buff = []
-            for x in self:
+            iterable = self._iterable.chunks() if isinstance(self._iterable, ChunkedTable) else self
+            for x in iterable:
                 if len(buff) == num_worker:
                     queue.put(await buff.pop(0))
                 buff.append(
-                    loop.run_in_executor(executor, _map_task(x, unary_op)))
+                    loop.run_in_executor(executor, self._map_task(x, unary_op)))
+            # for x in self:
```
Comment (reiase) [inline towhee/functional/mixins/parallel.py:244]: please remove unused code.

File: towhee/functional/mixins/column.py [hunk 3]
```diff
@@ -203,13 +203,16 @@ def cmap(self, unary_op):
...
+        if self.get_executor() is None:
+            if isinstance(self._iterable, ChunkedTable):
+                if not self.is_stream:
+                    tables = [WritableTable(self.__table_apply__(chunk, unary_op)) for chunk in self._iterable.chunks()]
+                else:
+                    tables = (WritableTable(self.__table_apply__(chunk, unary_op)) for chunk in self._iterable.chunks())
+                return self._factory(ChunkedTable(chunks = tables))
+            return self._factory(self.__table_apply__(self._iterable, unary_op))
+        else:
+            return self.pmap(unary_op)
```
Comment (reiase) [inline towhee/functional/mixins/column.py:215]: It seems that there is no test coverage for line 215

**Review by reiase — CHANGES_REQUESTED** [review]

File: towhee/functional/mixins/parallel.py (hunk 2 above)
Comment (Chiiizzzy) [reply towhee/functional/mixins/parallel.py]: Fixed

File: towhee/functional/mixins/parallel.py (hunk 1 above)
Comment (Chiiizzzy) [reply towhee/functional/mixins/parallel.py:268]: Shouldn't it be a new dc contains Writabletable? I've changed the return value of `_map_task` accordingly, i.e. return a Writabletable for each chunk, I suppose this is the expected behavior.

**Pull request merged**